dali/operators/generic/transpose/transpose.cc

```diff
@@ -12,25 +12,71 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "dali/kernels/common/transpose.h"
+#include "dali/core/static_switch.h"
 #include "dali/operators/generic/transpose/transpose.h"
+#include "dali/pipeline/data/views.h"
 
 namespace dali {
 
+#define TRANSPOSE_ALLOWED_TYPES                                                                    \
+  (bool, uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, float16, \
```

Comment:
Do we need to differentiate between types or just sizes of types (alignment)?

Reply:
I tried some alignment only but it's more complicated and listing all types  seems to better suit the compiler and it's a bit more performant.

Reply:
Good point. It would automatically handle any new data formats, should they appear.

Reply:
Actually, what would work best (?) would be to reinterpret the data as intXX based on sizeof(T). This preserves the natural element size and alignment as well as would reduce build time and binary size by instantiating less than half of the templates instantiated now.